packages/ckeditor5-dev-env/lib/release-tools/utils/transform-commit/transformcommitforsubrepository.js

```diff
@@ -5,31 +5,35 @@
 
 'use strict';
 
-const chalk = require( 'chalk' );
-const { logger } = require( '@ckeditor/ckeditor5-dev-utils' );
 const utils = require( './transform-commit-utils' );
 
-// A size of indent for a log. The number is equal to length of the log string:
-// '* 1234567 ', where '1234567' is a short commit id.
-const INDENT_SIZE = 10;
-
 /**
  * Parses a single commit:
  *   - displays a log when the commit has invalid format of the message,
  *   - filters out the commit if it should not be visible in the changelog,
  *   - makes links to issues and organizations on GitHub.
  *
- * @param {Commit} commit
- * @param {Object} context
- * @param {Boolean} context.displayLogs Whether to display the logs.
+ * Returns `undefined` if given commit should not be added to the changelog. This behavior can be changed
+ * using the `context.returnInvalidCommit` option.
+ *
+ * @param {Commit} rawCommit
+ * @param {Object} [context={}]
  * @param {Boolean} [context.returnInvalidCommit=false] Whether invalid commit should be returned.
- * @param {Object} context.packageData Content from the 'package.json' for given package.
- * @returns {Commit}
+ * @returns {Commit|undefined}
  */
-module.exports = function transformCommitForSubRepository( commit, context ) {
-	const log = logger( context.displayLogs ? 'info' : 'error' );
+module.exports = function transformCommitForSubRepository( rawCommit, context = {} ) {
+	// Let's clone the commit. We don't want to modify the reference.
+	const commit = Object.assign( {}, rawCommit );
+	commit.notes = [];
 
-	const hasCorrectType = utils.availableCommitTypes.has( commit.type );
+	for ( const note of rawCommit.notes ) {
+		commit.notes.push( Object.assign( {}, note ) );
+	}
+
+	// Copy the original `type` of the commit.
+	commit.rawType = commit.type;
```

Comment:
I'd use the `commit.rawType` a few times below instead of the `commit.type` since you change a little bit the meaning of the `commit.type` later.

Reply:
You can even do some swaps in the initial `Object.assign()`.